.github/workflows/update-docs-cli.yml

```diff
@@ -0,0 +1,77 @@
+# Copyright The Notary Project Authors.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: update-cli-docs
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'cmd/notation/**'
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-cli-docs:
+    name: Update CLI Documentation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+
+      - name: Set up Go
+        uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # v5.5.0
+        with:
+          go-version-file: 'go.mod'
+          check-latest: true
+
+      - name: Generate CLI documentation
```

Comment:
Where will the CLI reference doc to be generated? From the `cmd/notation/doc.go`, will the doc be generated in a new folder created in the `notation` repo?